### ISSUE TITLE
[CLOSED] Additional inputs

### ISSUE THREAD

**Issue by RobinPetit** [WizardCorp/WizardPoker#92]
<a href="https://github.com/StanIsAdmin"><img src="https://avatars.githubusercontent.com/u/15045096?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [StanIsAdmin](https://github.com/StanIsAdmin)**
_Friday Mar 11, 2016 at 11:18 GMT_
_Originally opened as https://github.com/ulb-infof209/Group5/issues/92_

---

For now, cards requiring additional inputs (attack a creature, steal a card from opponent) work in a strange and inconvenient way (IMO) : 
- The Client knows for a fact that a creature needs one input to perform an attack.
- The Client always expects a message from the server telling him how many additional inputs should be required to complete the use of a card/attack (some attacks could be made against multiple creatures, some spells apply to creatures so they must be selected)...

I think this is a strange behavior because the possible inputs should be only known to the server : if a creature is not allowed to attack the opponent player, if a creature can attack two opponent's creatures, if some other rules come into play here, they should be managed by the server only.
Basically, I think that ALL input requests should be sent from the server, specifying WHAT has to be selected. The Client simply sends a request to use a card or attack with a creature, and then receives the input requests whenever the server needs them to apply the attack/effects.
The alternative for a clearer mechanic would be for the Client to know which cards require which inputs, that could be stored in the database. But this would basically mean storing effects in the client database, which is not what is intended. AND then, we wouldn't be able to make this dynamic and effets-dependent. So I think the first option is best.
